engine/src/main/java/org/pentaho/di/trans/steps/script/ScriptAddedFunctions.java

```diff
@@ -0,0 +1,2532 @@
+           * End Bugfix
+           */
+        }
+      } catch ( Exception e ) {
+        throw new RuntimeException( e.toString() );
+      }
+    } else {
+      throw new RuntimeException( "The function call dateDiff requires 3 arguments." );
+    }
+  }
+
+  public static Object getNextWorkingDay( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+    // (Date dIn){
+    if ( ArgList.length == 1 ) {
+      try {
+        if ( isNull( ArgList[0] ) ) {
+          return null;
+        } else if ( isUndefined( ArgList[0] ) ) {
+          return undefinedValue;
+        }
+        java.util.Date dIn = (java.util.Date) ArgList[0];
+        Calendar startDate = Calendar.getInstance();
+        startDate.setTime( dIn );
+        startDate.add( Calendar.DATE, 1 );
+        while ( startDate.get( Calendar.DAY_OF_WEEK ) == Calendar.SATURDAY
+          || startDate.get( Calendar.DAY_OF_WEEK ) == Calendar.SUNDAY ) {
+          startDate.add( Calendar.DATE, 1 );
+        }
+        return startDate.getTime();
+      } catch ( Exception e ) {
+        throw new RuntimeException( e.toString() );
+      }
+    } else {
+      throw new RuntimeException( "The function call getNextWorkingDay requires 1 argument." );
+    }
+  }
+
+  public static Object dateAdd( ScriptEngine actualContext, Bindings actualObject, Object[] ArgList,
+    Object FunctionContext ) {
+    if ( ArgList.length == 3 ) {
+      try {
+        if ( isNull( ArgList, new int[] { 0, 1, 2 } ) ) {
+          return null;
+        } else if ( isUndefined( ArgList, new int[] { 0, 1, 2 } ) ) {
+          return undefinedValue;
+        }
+        java.util.Date dIn = (java.util.Date) ArgList[0];
+        String strType = ( (String) ArgList[1] ).toLowerCase();
+        int iValue = (Integer) ArgList[2];
+        Calendar cal = Calendar.getInstance();
+        cal.setTime( dIn );
+        if ( strType.equals( "y" ) ) {
+          cal.add( Calendar.YEAR, iValue );
+        } else if ( strType.equals( "m" ) ) {
+          cal.add( Calendar.MONTH, iValue );
+        } else if ( strType.equals( "d" ) ) {
+          cal.add( Calendar.DATE, iValue );
+        } else if ( strType.equals( "w" ) ) {
+          cal.add( Calendar.WEEK_OF_YEAR, iValue );
+        } else if ( strType.equals( "wd" ) ) {
+          int iOffset = 0;
+          while ( iOffset < iValue ) {
+            int day = cal.get( Calendar.DAY_OF_WEEK );
+            cal.add( Calendar.DATE, 1 );
+            if ( ( day != Calendar.SATURDAY ) && ( day != Calendar.SUNDAY ) ) {
+              iOffset++;
+            }
+          }
+        } else if ( strType.equals( "hh" ) ) {
+          cal.add( Calendar.HOUR, iValue );
+        } else if ( strType.equals( "mi" ) ) {
+          cal.add( Calendar.MINUTE, iValue );
+        } else if ( strType.equals( "ss" ) ) {
+          cal.add( Calendar.SECOND, iValue );
+        }
+        return cal.getTime();
+      } catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)